upstairs/src/lib.rs

```diff
@@ -3964,27 +3964,15 @@ impl Downstairs {
             kvec.sort_unstable();
 
             for id in kvec.iter() {
-                // Remove everything before this flush
+                // Remove everything before this flush (only valid if flushes
```

Comment:
Minor comment, I would word this as "Remove everything before this flush because flushes depend on everything and everything depends on flushes" Instead of the "only valid if" part.  If we change the code, then we should change the comment, the comment does not need the conditional.

Reply:
:+1:, done in f243621